### ISSUE TITLE
Experiencing connectivity issues

### ISSUE THREAD

**Issue by pumazi** [Connexions/cnx-deploy#243]
## Summary

This comes out of #242, but does not appear to be directly related. The content-copy-tool's requests seems to trigger this error after sword requests.

```pytb
Traceback (most recent call last):
  File "/var/lib/cnx/cnx-buildout/parts/zope2/lib/python/Shared/DC/ZRDB/Connection.py", line 71, in __setstate__
    try: self.connect(self.connection_string)
  File "/var/lib/cnx/cnx-buildout/eggs/Products.ZPsycopgDA-2.4.6_rhaptos-py2.4.egg/Products/ZPsycopgDA/DA.py", line 127, in connect
    self._v_database_connection.open()
  File "/var/lib/cnx/cnx-buildout/eggs/Products.ZPsycopgDA-2.4.6_rhaptos-py2.4.egg/Products/ZPsycopgDA/db.py", line 96, in open
    self.getconn()
  File "/var/lib/cnx/cnx-buildout/eggs/Products.ZPsycopgDA-2.4.6_rhaptos-py2.4.egg/Products/ZPsycopgDA/db.py", line 58, in getconn
    conn.set_session(isolation_level=int(self.tilevel))
ProgrammingError: set_session cannot be used inside a transaction
```

## Environment

dev & staging

**Comment by pumazi** [comment]
I'm not sure there is a correlation here or not...

> I'm observing an interesting pattern. I’ve cut the dev env down to 2 zclient instances (one on each server). The set_session issue flip flops between them. That is, one has a series of successes without showing that error while the other has a series of success showing the error. Then it flips.

**Comment by pumazi** [comment]
It's believed this was related to the Plone Hot Fix attribute access issues observed in #244. We were unable to reproduce the issue after the fix was put in place.
